hw/mcu/dialog/da1469x/src/hal_system.c

```diff
@@ -128,3 +107,20 @@ hal_reset_cause(void)
     return g_hal_reset_reason;
 #endif
 }
+
+static void
+da1469x_lpclk_settle_tmr_cb(void *arg)
+{
+    da1469x_clock_lp_xtal32k_switch();
+}
+
+void
+da1469x_lpclk_init(void)
+{
+    static struct hal_timer lpclk_settle_tmr;
+
+    da1469x_clock_lp_xtal32k_enable();
+
+    os_cputime_timer_init(&lpclk_settle_tmr, da1469x_lpclk_settle_tmr_cb, NULL);
+    os_cputime_timer_relative(&lpclk_settle_tmr, 8000 * 1000);
```

Comment:
Code will wait 8s for xtal32k to be ready, right?

Reply:
it will not explicitly wait, it will just defer switch (lp_clk will be rc32k during this time)